stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2514,6 +2519,78 @@ impl NakamotoChainState {
             tenure_height,
         )?;
 
+        if !block.is_first_mined() {
```

Comment:
What you have here is fine, but can you put it into its own method?  It'll make testing it easier.  Thanks! :pray:

Reply:
Yep -- I think this code block should be moved into its own function.